src/GobanSocket.ts

```diff
@@ -145,14 +145,11 @@ export class GobanSocket<
             }
 
             if (this.connected) {
-                this.send(
-                    "net/ping" as keyof SendProtocol,
-                    {
-                        client: Date.now(),
-                        drift: this.clock_drift,
-                        latency: this.latency,
-                    } as any,
-                );
+                this.send("net/ping", {
+                    client: Date.now(),
+                    drift: this.clock_drift,
+                    latency: this.latency,
+                } as DataArgument<SendProtocol["net/ping"]>);
```

Comment:
Unfortunately still a type assertion here.  *I think* the issue is that TypeScript has no good way of knowing DataArgument<T> is contravariant, and so doesn't realize this send call is legit for classes that extend `ClientToServer`.